src/core/load_balancing/pick_first/pick_first.cc

```diff
@@ -873,6 +872,8 @@ void PickFirst::SubchannelList::SubchannelData::OnConnectivityStateChange(
   // Otherwise, process connectivity state change.
   switch (*connectivity_state_) {
     case GRPC_CHANNEL_TRANSIENT_FAILURE: {
+      bool prev_seen_transient_failure =
+          std::exchange(seen_transient_failure_, true);
```

Comment:
question: do we have a guarantee that subchannels won't go from `CONNECTING` straight to `IDLE`?

Reply:
Yes.  The internal connectivity state API is non-lossy, so we'll always see the transition through TRANSIENT_FAILURE first, even if the backoff delay is very short.